netty/src/main/java/io/grpc/transport/netty/NettyClientTransport.java

```diff
@@ -90,6 +91,8 @@
   // We should not send on the channel until negotiation completes. This is a hard requirement
   // by SslHandler but is appropriate for HTTP/1.1 Upgrade as well.
   private Channel channel;
+  // TODO(jhump): should we use a SerializingExecutor
```

Comment:
The listener is always called from the same thread (the Netty EventLoop for this Channel) and the listener is not asynchronously, so synchronization concerns and ordering aren't an issue.